Initial project setup

isidora-b opened the pull request:
Initial spring boot app created, connection to mongodb tested

dunjagojkovic committed:
initial commit

Initial spring boot app created, connection to mongodb tested

tamastarjan commented on rbc/src/main/resources/application.yml:
```diff
@@ -0,0 +1,10 @@
+  application:
+    name: rbc
+  data:
+    mongodb:
+      host: localhost
+      port: 27017
+      database: library
+      username: library
+      password: library
```
Suggestion:
Use `spring.data.mongodb.uri: mongodb://<username>:<password>@<host>:<port>/<database>` property instead of individual properties. MongoDB connection strings (URIs) can be easily reused between different environments and tools.

teodoraBr commented on rbc/docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+  mongo-express:
+    image: mongo-express:latest
+    container_name: mongo-express
+    #restart: always
```
It is a good practice to resolve any commented-out code, either by removing it or uncommenting it, before merging. This helps maintain clarity and ensures a clean code. If there is a specific reason for the commented code to stay there it should be explained in the comment.

teodoraBr commented on rbc/docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+    ports:
+      - "27017:27017"
+    volumes:
+     - ./mongo-init.js:/docker-entrypoint-initdb.d/mongo-init.js:ro
```
You don't necessarily need the mongo-init.js file if you are already using the MONGO_INITDB_ROOT_USERNAME and MONGO_INITDB_ROOT_PASSWORD environment variables to set up the root user

teodoraBr commented on rbc/docker-compose.yml:
```diff
@@ -0,0 +1,33 @@
+      ME_CONFIG_BASICAUTH_PASSWORD: library
+      ME_CONFIG_MONGODB_URL: mongodb://library:library@mongo:27017/library?authSource=admin
+    ports:
+      - 8081:8081
```
Even though it works without quotes too, it is generally recommended to use quotes for all port mappings. This helps avoid potential parsing issues and and maintain consistency.

dunjagojkovic committed:
docker-compose fixed

Credentials for mongo db and mongo express changed, added quotes for ports and removing mongo-init.js file

dunjagojkovic committed:
update applicaton.yml

Using uri property instead of individual properties for mongodb

tamastarjan reviewed (approved):
Review: LGTM.
